File: 11-stacks-queues2/1-intro/main.cpp
// Advanced Stacks and Queues - C++ Example

#include <iostream>
#include <deque>

int main() {
    std::deque<int> dq;
    dq.push_back(1);       // add to right
    dq.push_front(2);      // add to left

    for (int val : dq) std::cout << val << " "; // prints: 2 1
    std::cout << std::endl;

    dq.pop_back();         // remove from right
    dq.pop_front();        // remove from left

    std::cout << "Size after pops: " << dq.size() << std::endl; // 0
    return 0;
}

File: 11-stacks-queues2/1-intro/main.py
# Advanced Stacks and Queues - Python Example

from collections import deque

# Demonstration of deque operations
dq = deque()
dq.append(1)       # add to right
dq.appendleft(2)   # add to left
print(dq)          # deque([2,1])

dq.pop()           # remove from right
dq.popleft()       # remove from left
print(dq)          # deque([])
